cabot/cabotapp/tests/tests_basic.py

```diff
@@ -328,6 +356,24 @@ def test_http_run_bad_resp(self):
         self.assertEqual(self.http_check.calculated_status,
                          Service.CALCULATED_FAILING_STATUS)
 
+    @patch('cabot.cabotapp.models.socket.create_connection', fake_tcp_success)
+    def test_tcp_success(self):
+        checkresults = self.tcp_check.statuscheckresult_set.all()
+        self.assertEqual(len(checkresults), 0)
+        self.tcp_check.run()
+        checkresults = self.tcp_check.statuscheckresult_set.all()
+        self.assertEqual(len(checkresults), 1)
+        self.assertTrue(self.tcp_check.last_result().succeeded)
+
+    @patch('cabot.cabotapp.models.socket.create_connection', fake_tcp_failure)
+    def test_tcp_failure(self):
+        checkresults = self.tcp_check.statuscheckresult_set.all()
+        self.assertEqual(len(checkresults), 0)
+        self.tcp_check.run()
+        checkresults = self.tcp_check.statuscheckresult_set.all()
+        self.assertEqual(len(checkresults), 1)
+        self.assertFalse(self.tcp_check.last_result().succeeded)
```

Comment:
For some reason I cannot correctly validate the error. The error found at self.tcp_check.last_result().error is shown to be u'', not 'timed out'. However, this shows the correct error on the web ui.